scripts/sysman/engineActivity.yml

```diff
@@ -0,0 +1,73 @@
+version: "1.7"
+name: $S_ENGINE_ACTIVITY_EXT_NAME
+value: '"$XS_extension_engine_activity"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Engine Activity Extension Version(s)"
+version: "1.7"
+name: $s_engine_activity_ext_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Extension properties related to Engine Groups"
+version: "1.7"
+class: $sEngine
+name: $s_engine_ext_properties_t
+base: $s_base_properties_t
+members:
+    - type: uint32_t
+      name: "countOfVirtualFunctionInstance"
+      desc: |
+            [out] Number of Virtual Function(VF) instances associated with engine to monitor the global utilization of hardware across all Virtual Function from a Physical Function (PF) instance.
+            These global and VF-by-VF views should provide engine group and individual engine level granularity.
+            This count represents the number of VF instances that are actively using the resource represented by the engine handle.
+details:
+    - "This structure may be passed to $sEngineGetProperties by having the pNext member of $s_engine_properties_t point at this struct."
+    - "Used for SRIOV per Virtual Function device utilization by $s_engine_group_t"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get activity stats for each Virtual Function (VF) associated with engine group."
+version: "1.7"
+class: $sEngine
+name: GetActivityExt
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_engine_handle_t
+      name: hEngine
+      desc: "[in] Handle for the component."
+    - type: uint32_t*
+      name: pCount
+      desc: |
+            [in,out] Pointer to the number of VF engine stats descriptors.
+              - if count is zero, the driver shall update the value with the total number of engine stats available.
+              - if count is greater than the total number of engine stats available, the driver shall update the value with the correct number of engine stats available.
+              - The count returned is the sum of number of VF instances currently available and the PF instance.
+    - type: $s_engine_stats_t*
+      name: pStats
+      desc: |
+            [in,out][optional][range(0, *pCount)] array of engine group activity counters.
+              - if count is less than the total number of engine stats available, then driver shall only retrieve that number of stats.
+              - the implementation shall populate the vector with user provided pCount number of VF engine stats followed by engine stat for PF.
+              - If VF has no activity value to report, then implementation shall use value of 0 to denote no activity.
```

Comment:
Clarification:
Would activity time be "0" ? If yes we could move this to $s_engine_stats_t description.
But I think the spec mentions that activity time is monotonically increasing. Please share if this will change.

Reply:
good point, i'll remove the lines from here and add note in the extension docs about interpreting zero activity

Reply:
fixed